test/arguments_test.cpp

```diff
@@ -120,7 +121,8 @@ TEST(ArgumentsTest, IpAddress)
     ASSERT_THROW(args.asIpAddress(), std::invalid_argument);
     ASSERT_THROW(args.asIpAddress(), std::invalid_argument);
     EXPECT_EQ(args.asIpAddress(), std::make_tuple(IpVer::v6, testArgs[5]));
-    EXPECT_EQ(args.asIpAddress(), std::make_tuple(IpVer::v6, testArgs[6]));
+    EXPECT_EQ(args.asIpAddress(), std::make_tuple(IpVer::v6, testArgs[5]));
```

Comment:
Are you sure it's `testArgs[5]` twice?

Reply:
Yes, I'm sure.
The first one compares the similar strings:
  `2001:db8:85a3::8a2e:370:7334` -> `2001:db8:85a3::8a2e:370:7334`
the second one compares extended by zeros string with the truncated one:
  `2001:0db8:85a3:0000:0000:8a2e:0370:7334` -> `2001:db8:85a3::8a2e:370:7334`

Reply:
Then you, probably should add a comment here explains your intention. It is not obvious from first look.

Reply:
done

Reply:
I don't get it. What truncates them? You have two absolutely identical lines and say that they do different things. Is there some magic happening between the two invocations?

Reply:
The 5-th argument is `2001:db8:85a3::8a2e:370:7334`. The  5-th test ensures that `asIpAddress()` didn't change it.
The 6-th argument is `2001:0db8:85a3:0000:0000:8a2e:0370:7334`. In fact, it is the same address like in the 5-th argument, but contains two zeroed words in the middle. 
The 6-th test ensures that `asIpAddress()` truncated these zeroed words and made the result equals the 5-th argument.

Reply:
Yes, I see that 6-th argument has extra zeroes, but after your patch lines 123 and 124 are __identical__ and both use testArgs[**5**]. I can understand that maybe in line 124 you now check that the test in line 123 hasn't changed the args or testArgs[5], but I don't see how and where you check that the 6th argument is properly handled because you now don't have testArgs[6] used anywhere.

Reply:
Each `args.asIpAddress()` call moves the arguments internal iterator.
That way, each following call works with a new argument taken from `testArgs`.

All these tests built on this idea.

Reply:
Ah, I see now. That is not obvious.